src/core/Akka.Tests/Actor/AskSpec.cs

```diff
@@ -203,6 +223,50 @@ public void Can_Ask_actor_inside_receive_loop()
             waitActor.Tell("ask");
             ExpectMsg("bar");
         }
+        
+        [Fact]
+        public void Generic_Ask_when_Failure_is_returned_should_throw_error_payload_and_preserve_stack_trace()
+        {
+            var actor = Sys.ActorOf<SomeActor>();
+            var aggregateException = Assert.Throws<AggregateException>(() =>
+            {
+                var result = actor.Ask<string>("throw", timeout: TimeSpan.FromSeconds(3)).Result;
```

Comment:
Guys, please let's not do result - it makes things complicated. We start doing AggregateException unwrapping, flattening and so on. Just by reading code I can't understand the level of nested exception...